File: grepper.rb
#!/usr/bin/env ruby

$LOAD_PATH << './lib'

require 'trollop'
require 'Grepper'

options = Trollop::options do
	opt :ignore_case, "Case-insensitive matching"
	opt :invert_match, "Inverted matching", :short => '-v'
	opt :line, "Display line number", :short => '-n'
	opt :filename, "Display file name"
end

grepper = Grepper::Grepper.new(
	:pattern => ARGV.shift,
	:is_case_sensitive => ! options[:ignore_case],
	:is_inverted => options[:invert_match],
	:show_file_name => options[:filename],
	:show_line_number => options[:line],
);

grepper.match(ARGF)

File: grepper.pl
#!/usr/bin/env perl

use strict;
use warnings;

use FindBin ();
use Getopt::Long ();

use lib "$FindBin::Bin/lib";
use Grepper;

my @files;
my $grepper = Grepper->new;

Getopt::Long::GetOptions(
	'' => sub { push @files, \*STDIN },
	'ignore-case|i!' => sub { $grepper->is_case_sensitive(! $_[1]) },
	'file=s' => \@files,
	'invert-match|v!' => sub { $grepper->is_inverted($_[1]) },
	'line|n!' =>  sub { $grepper->show_line_number($_[1]) },
	'filename!' => sub { $grepper->show_file_name($_[1]) },
);

$grepper->pattern(shift);
push @files, @ARGV;
@files = (\*STDIN) unless @files;

$grepper->match($_) for @files;
